src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1854,16 +1848,15 @@ def forward(
             input_values,
             output_attentions=output_attentions,
             output_hidden_states=output_hidden_states,
-            return_dict=return_dict,
+            return_dict=True,
         )
 
         hidden_states = outputs[0]
         hidden_states = self.dropout(hidden_states)
         logits = self.lm_head(hidden_states)
 
         if not return_dict:
-            output = (logits,) + outputs[2:]
```

Comment:
I think here we will need to revert to slicing the tuple as was done previously - IMO it's worth trying your solution where we index the tuple based on the settings of `output_hidden_states` / `output_attention`, since we sadly cannot force `return_dict=True`